### PR TITLE
Cmake and minmax

### PR DESCRIPTION
**Pull request by cboulay**
Reduce default buffer size for MacOS.
Fix Windows min/max problem according to @FlorianFranzen. Mirrors his commit: 6b5d9a3 plus changes to cbmex.
Modifed CMakeLists to make building static, mex, oct, test, hdf5 optional.
Make CMakeLists compatible with being an Added Project.
Eliminate warnings in MSVC 2015
Fixed output to dist dir in MSVC.
Copy Qt dlls into testcbsdk dir.
Made -fvisibility=hidden global to all projects. Added some placeholders to fix exports in static lib for Xcode. However, 'unexported' symbols from cbhwlib.h are used by cbmex.
Fixed some Clang warnings in cbsdk and cbmex.

Tested in MSVC 2015 x64 and x32. Tested in MacOS Sierra Xcode and GNU make.

### CODE REVIEW

File: cbmex/cbmex.cpp [hunk 1]
```diff
@@ -22,6 +22,7 @@
 
 */
 
+#include <algorithm>  // Use C++ default min and max implementation.
 #include "StdAfx.h"
```
Comment (dashesy) [inline cbmex/cbmex.cpp]: This (StdAfx) should be the first include (also in other places), otherwise Windows builds with MFC (like Blackrock's) will fail.

**Comment by dashesy** [comment]
Thanks for this! I will get your branch, test and merge soon

File: cbmex/cbmex.cpp (hunk 1 above)
Comment (cboulay) [inline cbmex/cbmex.cpp]: OK I can fix that. Do you want me to do a rebase and force push, or just add on another commit?

Comment (dashesy) [inline cbmex/cbmex.cpp]: Both are fine, 
The branch compiles fine on Linux, will merge after the change

**Comment by cboulay** [comment]
FYI, some of the motivation for the cmake changes were so I could include it as an ExternalProject in another app. See [here](https://github.com/cboulay/labstreaminglayer/blob/cerelink_app/Apps/BlackrockMicrosystems/Cerelink/CMakeLists.txt#L86-L122).

**Comment by dashesy** [comment]
It is good to have the option, and the default is as before.

**Pull request merged**